lib/Sema/CSSimplify.cpp

```diff
@@ -2029,6 +2116,36 @@ static bool fixExtraneousArguments(ConstraintSystem &cs,
       /*impact=*/numExtraneous * 2);
 }
 
+/// Determines if a failure with the given locator appears within a call or
+/// subscript argument list after the argument containing the code completion
+/// location.
+///
+/// \returns a locator pointing to the containing call/subscript if it meets
+/// these conditions and \c nullptr otherwise
+static ConstraintLocator *isInArgAfterCompletion(ConstraintLocator *loc,
+                                                 ConstraintSystem &cs) {
+  auto fixExpr = simplifyLocatorToAnchor(loc).dyn_cast<Expr *>();
+  auto commonAncestor = fixExpr;
+  auto fixChild = fixExpr;
+  while (commonAncestor && !cs.containsCodeCompletionLoc(commonAncestor)) {
+    fixChild = commonAncestor;
+    commonAncestor = cs.getParentExpr(commonAncestor);
+  }
+
+  if (!commonAncestor || commonAncestor == fixExpr)
+    return nullptr;
+
+  if (!commonAncestor || (!isa<CallExpr>(commonAncestor) && !isa<SubscriptExpr>(commonAncestor)))
```

Comment:
Should this check `commonAncestor->getArgs()` for null instead of matching parent types to e.g also handle ObjectLiteralExprs?